### PR TITLE
Add handling of Guid's in the JobDataMap

### PR DESCRIPTION
**Pull request by olicooper**
Resolves: https://github.com/quartznet/quartznet/issues/837

This will need testing prior to release.

### CODE REVIEW

**Comment by olicooper** [comment]
I'm unsure if the nullable Guid should be included as it doesn't quite fit with the other methods you have. Feel free to remove if it doesn't make sense.

File: src/Quartz/Util/StringKeyDirtyFlagMap.cs [hunk 1]
```diff
@@ -347,5 +356,39 @@ public virtual TimeSpan GetTimeSpan(string key)
...
+        /// </summary>
+        public virtual Guid GetGuid(string key)
+        {
+            object obj = this[key];
+
+            try
+            {
+                return (Guid)obj;
+            }
+            catch (Exception)
```
Comment (9Rune5) [inline src/Quartz/Util/StringKeyDirtyFlagMap.cs:371]: This try..catch adds 7 lines of code just to change "System.InvalidCastException: Specified cast is not valid." into "System.InvalidCastException: Identified object is not a Guid.".

That seems a bit overkill.

Comment (olicooper) [reply src/Quartz/Util/StringKeyDirtyFlagMap.cs]: Thanks for pointing this out, I have removed the try/catch blocks so it throws the original exception.

**Comment by olicooper** [comment]
I have compiled a version of this to use in a live project and I can confirm that using `JobDataMap.GetGuidValue("myGuid")` works as expected when using the JobStoreTX (MySqlConnector) provider.

**Comment by lahma** [comment]
Thank you for this too, great work! 👍

**Pull request merged**